app/src/main/java/com/example/socialapp/NextActivity.java

```diff
@@ -42,20 +50,19 @@ public void onClick(View view) {
         String code = data.getQueryParameter("code");
         getAccessToken(code);
     }
+
     private void getAccessToken(String code) {
-        LoginRepository loginRepository = new LoginRepositoryImpl();
-        Log.d("test", "Code = " + code);
-        Call<AccessTokenResponse> call = loginRepository.getAccessToken(code, "1848123932049765", "https://letsconnect.com/", "80d7301378a81e9758c559c097caf157");
+        Log.d(TAG, "Code = " + code);
+        Call<AccessTokenResponse> call = loginViewModel.getAccessToken(code, CLIENT_ID, REDIRECT_URI, CLIENT_SECRET);
         call.enqueue(new Callback<AccessTokenResponse>() {
             @Override
             public void onResponse(Call<AccessTokenResponse> call, Response<AccessTokenResponse> response) {
                 runOnUiThread(new Runnable() {
                     @Override
                     public void run() {
                         String accessToken = response.body().getAccessToken();
-                        SharedPreferences.Editor editor = getSharedPreferences("LetsConnect", MODE_PRIVATE).edit();
-                        editor.putString("accessToken", accessToken);
-                        editor.apply();
+                        SharedPreferenceHelper.setSharedPreferenceString(NextActivity.this, PREF_KEY_ACCESS_TOKEN, accessToken);
+                        SharedPreferenceHelper.getSharedPreferenceString(NextActivity.this, PREF_KEY_ACCESS_TOKEN, "");
```

Comment:
Why we need this line, if we don't want to fethch token here?

Reply:
Implemented